staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/filter.go

```diff
@@ -78,6 +78,9 @@ func (a *evaluationActivation) Parent() interpreter.Activation {
 func (c *filterCompiler) Compile(expressionAccessors []ExpressionAccessor, options OptionalVariableDeclarations, perCallLimit uint64) Filter {
 	compilationResults := make([]CompilationResult, len(expressionAccessors))
 	for i, expressionAccessor := range expressionAccessors {
+		if expressionAccessor == nil {
```

Comment:
this is surprising... is leaving zero-value entries in compilationResults going to break expectations anywhere we send this list?

Reply:
More info: This case is made special for messageExpression only since it could be unset and we wanna it to be a 1:1 mapping to validations[i].expression. For the rest of cases, the cel expression is expected to be always non-nil(It is guaranteed from api level and code impl).

Reply:
Thank you @cici37 for explanation. Here this special handling is an unfortunate compromise of the interface changes made in https://github.com/kubernetes/kubernetes/pull/115816 . Allowing nil as placeholders is least impactful on the ongoing development like https://github.com/kubernetes/kubernetes/pull/116261 that rely on the interfaces. I will send another refactor once all affecting PRs are merged, after this release.

Reply:
ok, as long as you've swept call sites / implementations to make sure they're all expecting the nil items, a refactor follow-up can be ok

Reply:
Yes these are all manually examined and fixed.

Reply:
Could we add test case for this?